napari/utils/theme.py

```diff
@@ -317,20 +353,19 @@ def rebuild_theme_settings():
     canvas='white',
 )
 
-register_theme('dark', DARK)
-register_theme('light', LIGHT)
+register_theme('dark', DARK, "builtin")
+register_theme('light', LIGHT, "builtin")
 
 
 # this function here instead of plugins._npe2 to avoid circular import
 def _install_npe2_themes(_themes):
     import npe2
 
-    for theme in npe2.PluginManager.instance().iter_themes():
-        # `theme.type` is dark/light and supplies defaults for keys that
-        # are not provided by the plugin
-        d = _themes[theme.type].dict()
-        d.update(theme.colors.dict(exclude_unset=True))
-        register_theme(theme.id, d)
+    for mf in npe2.PluginManager.instance().iter_manifests(disabled=False):
+        for theme in mf.contributions.themes or ():
+            d = _themes[theme.type].dict()
+            d.update(theme.colors.dict(exclude_unset=True))
+            register_theme(theme.id, d, mf.name)
```

Comment:
A bit hard to follow, could you use more verbose variable names?

Reply:
could you check now?